autogen/oai/bedrock.py

```diff
@@ -0,0 +1,602 @@
+
+        # Parameters vary based on the model used.
+        # As we won't cater for all models and parameters, it's the developer's
+        # responsibility to implement the parameters and they will only be
+        # included if the developer has it in the config.
+        #
+        # Important:
+        # No defaults will be used (as they can vary per model)
+        # No ranges will be used (as they can vary)
+        # We will cover all the main parameters but there may be others
+        # that need to be added later
+        #
+        # Here are some pages that show the parameters available for different models
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-text.html
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-anthropic-claude-text-completion.html
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-cohere-command-r-plus.html
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-meta.html
+        # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-mistral-chat-completion.html
+
+        # Here are the possible "base" parameters and their suitable types
+        base_parameters = [["temperature", (float, int)], ["topP", (float, int)], ["maxTokens", (int)]]
+
+        for param_name, suitable_types in base_parameters:
+            if param_name in params:
+                base_params[param_name] = validate_parameter(
+                    params, param_name, suitable_types, False, None, None, None
+                )
+
+        # Here are the possible "model-specific" parameters and their suitable types, known as additional parameters
+        additional_parameters = [
+            ["top_p", (float, int)],
+            ["top_k", (int)],
+            ["k", (int)],
+            ["seed", (int)],
+        ]
+
+        for param_name, suitable_types in additional_parameters:
+            if param_name in params:
+                additional_params[param_name] = validate_parameter(
+                    params, param_name, suitable_types, False, None, None, None
+                )
+
+        # Streaming
+        if "stream" in params:
+            self._streaming = params["stream"]
+        else:
+            self._streaming = False
+
+        # For this release we will not support streaming as many models do not support streaming with tool use
+        if self._streaming:
+            warnings.warn(
+                "Streaming is not currently supported, streaming will be disabled.",
+                UserWarning,
+            )
+            self._streaming = False
+
+        return base_params, additional_params
+
+    def create(self, params):
+        """Run Amazon Bedrock inference and return AutoGen response"""
+
+        # Establish Bedrock config and session from config parameters
+        self.parse_config_params(params)
```

Comment:
This is a good point - the `init` only supports the `api_key`, so I thought it may be cleaner to be able to parse it in the parameters.

The other option is to do what is done in the Anthropic class (which supports Anthropic through Bedrock). In the client.py class this is the code it uses:
```
            elif api_type is not None and api_type.startswith("anthropic"):
                if "api_key" not in config:
                    self._configure_openai_config_for_bedrock(config, openai_config)
                if anthropic_import_exception:
                    raise ImportError("Please install `anthropic` to use Anthropic API.")
                client = AnthropicClient(**openai_config)
                self._clients.append(client)
```

... where it uses a specific function to parse parameters for bedrock.

We could reuse this, let me know what you think.

Reply:
@marklysze I think the Anthropic class like option is good.

Reply:
Okay, let me update to align with that class, thanks!

Reply:
I've updated to align with the Anthropic client class parameters, same names and function.

For testing, parameters are now:
- aws_region (mandatory)
- aws_access_key (or environment variable: AWS_ACCESS_KEY)
- aws_secret_key (or environment variable: AWS_SECRET_KEY)
- aws_session_token (or environment variable: AWS_SESSION_TOKEN)
- aws_profile_name